examples/SDG1.ipynb

```diff
@@ -0,0 +1,1336 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 73,
+      "metadata": {
+        "id": "8pzwvE-YQPtU"
+      },
+      "outputs": [],
+      "source": [
+        "from openai import OpenAI\n",
+        "import re\n",
+        "import numpy as np\n",
+        "import pandas as pd\n",
+        "from sklearn.cluster import KMeans\n",
+        "import matplotlib.pyplot as plt\n",
+        "import json\n",
+        "import matplotlib"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "B8eAx4-JxaZB"
+      },
+      "source": [
+        "### 1. CSV with a structure prompt\n",
+        "Here we create data in the simplest way. You can quickly generate data by addressing 3 key points: telling it the format of the data (CSV), the schema, and useful information regarding how columns relate (the LLM will be able to deduce this from the column names but a helping hand will improve performance)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "dqbvepd0n4vS",
+        "outputId": "8735cacc-baa5-463e-938c-783e6b508b00"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "id,house size (m^2),house price,location,number of bedrooms\n",
+            "1,100,300000,New York,2\n",
+            "2,120,350000,New York,3\n",
+            "3,80,250000,New York,2\n",
+            "4,150,400000,Los Angeles,3\n",
+            "5,200,500000,Los Angeles,4\n",
+            "6,160,320000,Chicago,3\n",
+            "7,80,200000,Chicago,2\n",
+            "8,200,600000,San Francisco,4\n",
+            "9,240,700000,San Francisco,5\n",
+            "10,100,270000,Chicago,2\n"
+          ]
+        }
+      ],
+      "source": [
+        "question = \"\"\"\n",
+        "Create a CSV file with 10 rows of housing data.\n",
+        "Each row should include the following fields:\n",
+        " - id (incrementing integer starting at 1)\n",
+        " - house size (m^2)\n",
+        " - house price\n",
+        " - location\n",
+        " - number of bedrooms\n",
+        "\n",
+        "Make sure that the numbers make sense (i.e. more rooms is usually bigger size, more expensive locations increase price. more size is usually higher price etc. make sure all the numbers make sense). Also only respond with the CSV.\n",
+        "\"\"\"\n",
+        "\n",
+        "response = client.chat.completions.create(\n",
+        "  model=\"gpt-4-0125\",\n",
+        "  messages=[\n",
+        "    {\"role\": \"system\", \"content\": \"You are a helpful assistant designed to generate synthetic data.\"},\n",
+        "    {\"role\": \"user\", \"content\": question}\n",
+        "  ]\n",
+        ")\n",
+        "res = response.choices[0].message.content\n",
+        "print(res)"
```

Comment:
try `display(res)`?

Reply:
I tried displaying instead of printing but found it to be less pretty. Is it fine to keep as print in the current state?